dl1_data_handler/reader.py

```diff
@@ -456,6 +456,15 @@ def __getitem__(self, idx):
 
         return example
 
+
+    # Convert a possibly-nested sequence to a tuple
+    def _totuple(self, a):
+        try:
+            return tuple(self._totuple(i) for i in a)
+        except TypeError:
+            return a
```

Comment:
```suggestion
            return tuple(a)
```

Reply:
That was needed for me (for breakdown by 'particletype'). Can you please double check?

To breakdown by 'direction', it was breaking in `log_examples()` from the ctlearn utils as mentioned in [CTLearn PR 132](https://github.com/ctlearn-project/ctlearn/pull/132).

Reply:
I fixed the problem instead by converting the numpy array to a scalar. That is cleaner since otherwise the scalar labels would be wrapped as a tuple.

Reply:
Great! It's working fine! We probably want to have this fix on top of dl1dh v0.8.2, i.e. releasing dl1dh v0.8.3, to incorporate with ctlearn v0.5.0? Currently this PR want to merge into the master, which is dl1dh >v0.9.0.

Reply:
Great. I'll merge this PR into master now, and would you be able to create a v0.8.3 release using this branch?